tests/integration/models/os.spec.coffee

```diff
@@ -0,0 +1,271 @@
+
+		describe 'given a valid device slug', ->
+
+			it 'should eventually be a valid Date instance', ->
+				promise = resin.models.os.getLastModified('raspberry-pi')
+				m.chai.expect(promise).to.eventually.be.an.instanceof(Date)
+
+			it 'should eventually be a valid Date instance if passing a device type alias', ->
+				promise = resin.models.os.getLastModified('raspberrypi')
+				m.chai.expect(promise).to.eventually.be.an.instanceof(Date)
+
+			it 'should be able to query for a specific version', ->
+				promise = resin.models.os.getLastModified('raspberrypi', '1.24.0')
+				m.chai.expect(promise).to.eventually.be.an.instanceof(Date)
+
+			it 'should be able to query for a version containing a plus', ->
+				promise = resin.models.os.getLastModified('raspberrypi', '2.0.0+rev2')
+				m.chai.expect(promise).to.eventually.be.an.instanceof(Date)
+
+		describe 'given an invalid device slug', ->
+
+			it 'should be rejected with an error message', ->
+				promise = resin.models.os.getLastModified('foo-bar-baz')
+				m.chai.expect(promise).to.be.rejectedWith('No such device type')
+
+	describe 'resin.models.os.download()', ->
+		return if IS_BROWSER
+
+		rindle = require('rindle')
+		tmp = require('tmp')
+		fs = Promise.promisifyAll(require('fs'))
+
+		describe 'given a valid device slug', ->
+
+			it 'should contain a valid mime property', ->
+				resin.models.os.download('parallella').then (stream) ->
+					m.chai.expect(stream.mime).to.equal('application/octet-stream')
+
+			it 'should contain a valid mime property if passing a device type alias', ->
+				resin.models.os.download('raspberrypi').then (stream) ->
+					m.chai.expect(stream.mime).to.equal('application/octet-stream')
+
+			it 'should be able to download the image', ->
+				tmpFile = tmp.tmpNameSync()
+				resin.models.os.download('parallella').then (stream) ->
+					stream.pipe(fs.createWriteStream(tmpFile))
+				.then(rindle.wait)
+				.then ->
+					return fs.statAsync(tmpFile)
+				.then (stat) ->
+					m.chai.expect(stat.size).to.not.equal(0)
+				.finally ->
+					fs.unlinkAsync(tmpFile)
+
+		describe 'given an invalid device slug', ->
+
+			it 'should be rejected with an error message', ->
+				promise = resin.models.os.download('foo-bar-baz')
+				m.chai.expect(promise).to.be.rejectedWith('No such device type')
+
+	describe 'when logged in as a user with a single application', ->
+
+		givenLoggedInUser()
+
+		beforeEach ->
+			resin.models.application.create('FooBar', 'raspberry-pi').then (application) =>
+				@application = application
+
+		describe 'resin.models.os.getConfig()', ->
+			beforeEach ->
+				resin.models.application.create('TestApp', 'raspberry-pi').then (application) =>
+					@application = application
+
+			it 'should be able to get an application config by id', ->
+				promise = resin.models.os.getConfig(@application.id)
+				Promise.all [
+					m.chai.expect(promise).to.eventually.have.property('applicationId')
```

Comment:
This pattern is repeated so often it feels like it would make sense to have
```coffee
expectProperty = (promise, prop) ->
	m.chai.expect(promise).to.eventually.have.property(prop)
...
expectProperty(promise, 'apiKey')
expectProperty(promise, 'userId')
...